web/pages/link.tsx

```diff
@@ -0,0 +1,322 @@
+import { Claim, Manalink } from 'common/manalink'
+import { formatMoney } from 'common/util/format'
+import { Col } from 'web/components/layout/col'
+import { Page } from 'web/components/page'
+import { SEO } from 'web/components/SEO'
+import { Title } from 'web/components/title'
+import { useUser } from 'web/hooks/use-user'
+import { createManalink, useUserManalinks } from 'web/lib/firebase/manalinks'
+import { fromNow } from 'web/lib/util/time'
+import { useManalinkTxns } from 'web/lib/firebase/txns'
+import { useUserById } from 'web/hooks/use-users'
+import { Txn } from 'common/txn'
+import { Avatar } from 'web/components/avatar'
+import { RelativeTimestamp } from 'web/components/relative-timestamp'
+import { UserLink } from 'web/components/user-page'
+import { ManalinkCard, ManalinkInfo } from 'web/components/manalink-card'
+
+import Textarea from 'react-expanding-textarea'
+
+import dayjs from 'dayjs'
+import customParseFormat from 'dayjs/plugin/customParseFormat'
+dayjs.extend(customParseFormat)
+
+function getLinkUrl(slug: string) {
+  return `${location.protocol}//${location.host}/link/${slug}`
+}
+
+export default function LinkPage() {
+  const user = useUser()
+  const [newManalink, setNewManalink] = useState<ManalinkInfo>({
+    expiresTime: null,
+    amount: 100,
+    maxUses: 5,
+    uses: 0,
+    message: '',
+  })
+  const links = useUserManalinks(user?.id ?? '')
+  const manalinkTxns = useManalinkTxns(user?.id ?? '')
+  const outstandingLinks = links.filter(
+    (l) =>
+      (l.maxUses == null || l.claimedUserIds.length < l.maxUses) &&
+      (l.expiresTime == null || l.expiresTime > Date.now())
+  )
+
+  if (user == null) {
+    return null
+  }
+
+  return (
+    <Page>
+      <SEO
+        title="Create a manalink"
+        description="Send mana to anyone via link!"
+        url="/send"
+      />
+      <Col className="w-full px-8">
+        <Title text="Create a manalink" />
+        <p>
+          You can use manalinks to send mana to other people, even if they
+          don&apos;t yet have a Manifold account.
+        </p>
+        <form
+          className="my-5"
+          onSubmit={async (e) => {
+            e.preventDefault()
+            await createManalink({
+              fromId: user.id,
+              amount: newManalink.amount,
+              expiresTime: newManalink.expiresTime,
+              maxUses: newManalink.maxUses,
+              message: newManalink.message,
+            })
+          }}
+        >
+          <div className="flex flex-row flex-wrap gap-x-5 gap-y-2">
+            <div className="form-control flex-auto">
+              <label className="label">Amount</label>
+              <input
+                className="input"
+                type="number"
+                value={newManalink.amount}
+                onChange={(e) =>
+                  setNewManalink((m) => {
+                    return { ...m, amount: parseInt(e.target.value) }
+                  })
```

Comment:
[Not important] Since all the ways of updating a mana link follow this structure, a neat pattern is to replace the useState with a simple useReducer:
```
const [newManaLink, updateNewManaLink] = useReducer((m, update) => ({ ...m, ...update }), (default value here))
```
then on these lines you just update like `updateNewManaLink({ amount: parseInt(e.target.value) })`